apps/ai-content-generator/src/utils/dialog/supported-fields/supportedFieldsHelpers.ts

```diff
@@ -28,28 +43,55 @@ const formatField = (field: ContentFields, entry: EntryProps, targetLocale: stri
 };
 
 /**
- * A reducer function that checks if a field is supported, whether it has content,
- * and assigns the field to the correct column.
+ * A reducer function that checks if a field is a supported type, iterates through its supported locales,
+ * determines whether it has content, and assigns the field to the correct column.
  *
  * TODO: Handle the case where the field is empty
  * TODO: Support storing fields for both Source and Output
  * @param entry
  * @param supportedFields
- * @param locale
+ * @param fieldLocales
+ * @param localeNames
+ * @param defaultLocale
  * @returns
  */
-const isSupported = (entry: EntryProps, supportedFields: SupportedFieldTypes[], locale: string) => {
+const isSupported = (
+  entry: EntryProps,
+  supportedFields: SupportedFieldTypes[],
+  fieldLocales: FieldLocales,
+  localeNames: LocaleNames,
+  defaultLocale: string
+) => {
   return (fieldAcc: SupportedFieldsOutput, field: ContentFields) => {
-    const isSupported = supportedFields.includes(field.type as SupportedFieldTypes);
-    const hasContent = entry.fields[field.id]?.[locale];
-    const formattedField = formatField(field, entry, locale);
+    const isSupportedFieldType = supportedFields.includes(field.type as SupportedFieldTypes);
 
-    if (isSupported && hasContent) {
-      fieldAcc.supportedFieldsWithContent.push(formattedField);
-    }
+    if (isSupportedFieldType) {
+      const fieldsWithContent = [] as Field[];
+      const allFields = [] as Field[];
+
+      fieldLocales[field.id].forEach((locale) => {
+        const hasContent = entry.fields[field.id]?.[locale];
+        const formattedField = formatField(field, entry, locale, localeNames, defaultLocale);
+
+        if (hasContent) {
+          fieldsWithContent.push(formattedField);
+        }
+        allFields.push(formattedField);
+      });
+
+      // Sort the locales so that default is first, then alphabetically by language name
+      const sortedFieldsWithContent = orderBy(
```

Comment:
I feel this can be done with the vanilla sort, as opposed to bringing in lodash here:

```js
// another helper function
const sortLanguages =  (a, b) => {
  if (a.isDefaultLocale) {
    // send default to top
    return -1;
  }

  if (a.language > b.language) {
    return 1
  }

  if (a.language < b.language) {
    return -1
  }
  
  return 0;
}

// inside isSupported
...

const sortedFieldsWithContext = fieldsWithContext.sort(sortLanguages);
const sortedAllFields = allFields.sort(sortLanguages)
```

Reply:
I removed lodash and made this change; had to update the logic a little bit from what you proposed here to get the sorting working right, but it's now implemented with vanilla sort!

Reply:
Had to pull this down to mess around with it a bit more. I riffed on your implementation a bit to cover the edge case of if b is the default locale. Also found `localeCompare` which simplifies this a bit more!

```js
      const sortedAllFields = allFields.sort((a: Field, b: Field) => {
        // Default should be listed first
        if (a.isDefaultLocale) {
          return -1;
        }

        // Default should be listed first
        if (b.isDefaultLocale) {
          return 1;
        }

        return a.name.localeCompare(b.name);
      });
```